File: app/src/main/java/com/example/alarm/room/AlarmDbEntity.java
package com.example.alarm.room;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

import com.example.alarm.model.Alarm;

@Entity(
        tableName = "alarms",
        indices = {
                @Index(value = {"time_hours", "time_minutes"}, unique = true)
        }
)
public class AlarmDbEntity {
    @PrimaryKey(autoGenerate = true) public long id;
    @ColumnInfo(name = "time_hours") public int timeHours;
    @ColumnInfo(name = "time_minutes") public int timeMinutes;
    @NonNull public String name;
    @ColumnInfo(name = "enabled", defaultValue = "0") public boolean enabled;

    public AlarmDbEntity(long id, int timeHours, int timeMinutes, @NonNull String name, boolean enabled) {
        this.id = id;
        this.timeHours = timeHours;
        this.timeMinutes = timeMinutes;
        this.name = name;
        this.enabled = enabled;
    }

    public Alarm toAlarm() {
        Alarm alarm = new Alarm(id);
        alarm.setTimeHours(timeHours);
        alarm.setTimeMinutes(timeMinutes);
        alarm.setName(name);
        alarm.setEnabled(enabled);
        return alarm;
    }

    public static AlarmDbEntity fromUserInput(Alarm alarm) {
        return new AlarmDbEntity(alarm.getId(), alarm.getTimeHours(), alarm.getTimeMinutes(), alarm.getName(), alarm.getEnabled());
    }
}

File: app/src/main/java/com/example/alarm/room/AlarmDbEntity.kt
package com.example.alarm.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import com.example.alarm.model.Alarm

@Entity(
    tableName = "alarms",
    indices = [
        Index("time_hours", "time_minutes", unique = true),
    ]
)
data class AlarmDbEntity(
    @PrimaryKey(autoGenerate = true) val id: Long,
    @ColumnInfo(name = "time_hours", collate = ColumnInfo.NOCASE) var timeHours: Int,
    @ColumnInfo(name = "time_minutes", collate = ColumnInfo.NOCASE) var timeMinutes: Int,
    var name: String,
    var enabled: Boolean
    ) {

    fun toAlarm(): Alarm = Alarm(
        id = id,
        timeHours = timeHours,
        timeMinutes = timeMinutes,
        name = name,
        enabled = enabled
    )

    companion object {
        fun fromUserInput(alarm: Alarm): AlarmDbEntity = AlarmDbEntity(
            id = 0,
            timeHours = alarm.timeHours,
            timeMinutes = alarm.timeMinutes,
            name = alarm.name,
            enabled = alarm.enabled
        )
    }
}